fetch_crypto_prices.py

```diff
@@ -2,24 +2,69 @@
 import json
 import os
 from datetime import datetime
+import logging
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
 
 # Get BASE_URL from environment variable
 BASE_URL = os.getenv('BASE_URL')
 
-def fetch_crypto_prices():
-    url = f"{BASE_URL}/simple/price?ids=dai,ethereum,bitcoin&vs_currencies=usd"
-    response = requests.get(url)
-    if response.status_code == 200:
+# Setup retry strategy for network issues
+retry_strategy = Retry(
+    total=3,  # Retry 3 times
+    backoff_factor=1,  # Wait between retries (exponential backoff)
+    status_forcelist=[429, 500, 502, 503, 504],  # Retry on these HTTP status codes
+    method_whitelist=["GET"]
+)
```

Comment:
If we're going to use this retry strategy, please clean up the retry mechanism in the github actions workflow file